test/e2e/common/node/runtimeclass.go

```diff
@@ -375,12 +374,15 @@ func createRuntimeClass(ctx context.Context, f *framework.Framework, name, handl
 	return rc.GetName()
 }
 
-func expectPodRejection(ctx context.Context, f *framework.Framework, pod *v1.Pod) {
-	_, err := f.ClientSet.CoreV1().Pods(f.Namespace.Name).Create(ctx, pod, metav1.CreateOptions{})
-	gomega.Expect(err).To(gomega.HaveOccurred(), "should be forbidden")
-	if !apierrors.IsForbidden(err) {
-		framework.Failf("expected forbidden error, got %#v", err)
-	}
+func expectPodRejection(ctx context.Context, f *framework.Framework, rcName string) {
+	gomega.Eventually(ctx, func() error {
+		pod := e2eruntimeclass.NewRuntimeClassPod(rcName)
+		_, err := f.ClientSet.CoreV1().Pods(f.Namespace.Name).Create(ctx, pod, metav1.CreateOptions{})
```

Comment:
```suggestion
		_, err := f.ClientSet.CoreV1().Pods(f.Namespace.Name).Create(ctx, pod, metav1.CreateOptions{DryRun: []string{"All"}})
```

> You could use dry-run to avoid spamming a bunch of pods that need to be cleaned up.

Reply:
updated. https://github.com/kubernetes/kubernetes/pull/130009/commits/9fa39276e7d4e7d751efd419e814ea8373a97874